projects/extension/build.py

```diff
@@ -116,13 +186,33 @@ def incremental_sql_files() -> list[Path]:
 def check_incremental_sql_files(paths: list[Path]) -> None:
     prev = 0
     for path in paths:
-        this = int(path.name[0:3])
-        if this != prev + 1:
+        check_sql_file_name(path)
+        this = file_number(path)
+        if this < 900 and this != prev + 1:
             print(
                 f"incremental sql files must be strictly ordered. this: {this} prev: {prev}",
                 file=sys.stderr,
             )
             sys.exit(1)
+        if this >= 900 and this == prev:  # allow gaps in pre-production scripts
+            print(
+                f"incremental sql files must not have duplicate numbers. this: {this} prev: {prev}",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        ff = parse_feature_flag(path)
+        if this < 900 and ff:
+            print(
+                f"incremental sql files under 900 must be NOT gated by a feature flag: {path.name}",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+        if this >= 900 and not ff:
+            print(
+                f"incremental sql files over 899 must be gated by a feature flag: {path.name}",
+                file=sys.stderr,
+            )
+            sys.exit(1)
```

Comment:
Nit, would it make sense to extract this block of code extracted to a common func to be used in both `check_incremental_sql_files` and `check_idempotent_sql_files`? Seems to be checking the same conditions.

Reply:
done